### PR TITLE
Package update and enhancements/fixes in YGOT, and Request Binder

### PR DESCRIPTION
**Pull request by mbalachandar**
YGOT:-
1. To improve the performance of the Unmarshal and EmitJSON methods.
      - Caching the schema information to path instead of finding the schema for the path every time.
      - Disabled the validation of the response while marshaling the ygot object.
      - Added the debug check for the debug logs to get generated/printed only when the debug flag is enabled
2. Fixes in the Unmarshal method for the node leaf-list, and leaf contains union which has enum as one of its type.
3. Added changes in the Unmarshal method to throw an error if the request payload has the state information.

Request Binder:-
1. Added the method validateObjectType - to check and throw an error if the given request payload contains state information.
2. Added the changes not to validate the request payload if the model of the payload is sonic yang, since CVL validates the sonic yang model request

Package update:-
1. Updated the ygot package to the version v0.7.1
2. Updated the goyang package to the version v0.0.0-20200309174518-a00bece872fc
3. Updated the gnmi package to the version v0.0.0-20200307010808-e7106f7f5493

### CODE REVIEW

File: patches/ygot/ygot.patch [hunk 1]
```diff
@@ -1,6 +1,58 @@
+diff -ruN ygot-dir-orig/ygot/generator/generator.go ygot-dir/ygot/generator/generator.go
+--- ygot-dir-orig/ygot/generator/generator.go	2020-06-21 15:35:13.777667000 -0700
++++ ygot-dir/ygot/generator/generator.go	2020-06-21 15:27:34.931674000 -0700
+@@ -109,7 +109,15 @@
+ }
+ 
+ // writeIfNotEmpty writes the string s to b if it has a non-zero length.
+-func writeIfNotEmpty(b io.StringWriter, s string) {
++// commented out the below method, since "io.StringWriter" is introduced in go 1.12
++//func writeIfNotEmpty(b io.StringWriter, s string) {
```
Comment (renukamanavalan) [inline patches/ygot/ygot.patch:10]: Please remove functions that are not in use. We could bring it back, when it is required in future.

File: patches/ygot/ygot.patch [hunk 2]
```diff
@@ -1,6 +1,58 @@
...
+ 	"github.com/openconfig/ygot/util"
+ 	"github.com/openconfig/ygot/ygot"
+@@ -42,7 +42,15 @@
+ )
+ 
+ // WriteIfNotEmpty writes the string s to b if it has a non-zero length.
+-func WriteIfNotEmpty(b io.StringWriter, s string) {
++// commented out the below method, since we don't support go 1.12 currently
++//func WriteIfNotEmpty(b io.StringWriter, s string) {
++//	if len(s) != 0 {
```
Comment (renukamanavalan) [inline patches/ygot/ygot.patch:43]: Same. Please remove these commented lines.

File: patches/ygot/ygot.patch [hunk 3]
```diff
@@ -71,93 +205,224 @@ diff -ruN ygot-dir-orig/ygot/util/reflect.go ygot-dir/ygot/util/reflect.go
...
- 		var ok bool
- 		s, ok = s.Dir[p[0]]
- 		if !ok {
-@@ -261,10 +273,13 @@
- 		return nil, nil
+-	if err := s.Validate(vopts...); !skipValidation && err != nil {
+-		return "", fmt.Errorf("validation err: %v", err)
+-	}
++//	var (
++//		vopts          []ValidationOption
```
Comment (renukamanavalan) [inline patches/ygot/ygot.patch:391]: Same. Please remove commented code.

File: patches/ygot/ygot.patch (hunk 2 above)
Comment (mbalachandar) [reply patches/ygot/ygot.patch]: Addressed it. Thanks..

**Pull request merged**